src/google/protobuf/compiler/objectivec/objectivec_helpers.h

```diff
@@ -87,8 +87,8 @@ string FileClassName(const FileDescriptor* file);
 
 // These return the fully-qualified class name corresponding to the given
 // descriptor.
-string ClassName(const Descriptor* descriptor);
-string ClassName(const Descriptor* descriptor, string* out_suffix_added);
+string LIBPROTOC_EXPORT ClassName(const Descriptor* descriptor);
+string LIBPROTOC_EXPORT ClassName(const Descriptor* descriptor, string* out_suffix_added);
 string EnumName(const EnumDescriptor* descriptor);
```

Comment:
Yeah, please add LIBPROTOC_EXPORT to all public functions in charp_names.h as well. Right now the GetOutputFile() function in csharp_names.h is also missing it.